### PR TITLE
Bump to volcano 0.4.0

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Depend on #886

### CODE REVIEW

**Comment by liyinan926** [comment]
Please rebase and resolve the conflict.

File: pkg/webhook/webhook.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ func (wh *WebHook) selfRegistration(webhookConfigName string) error {
 		},
 	}
 
-	mutatingWebhook := v1beta1.Webhook{
+	mutatingWebhook := v1beta1.MutatingWebhook{
```
Comment (liyinan926) [inline pkg/webhook/webhook.go:381]: So it has been renamed in a newer version of the k8s API?

File: go.mod [hunk 2]
```diff
@@ -3,53 +3,62 @@ module github.com/GoogleCloudPlatform/spark-on-k8s-operator
...
+	k8s.io/kubectl v0.0.0
+	k8s.io/kubernetes v1.16.2
 	k8s.io/utils v0.0.0-20200414100711-2df71ebbae66 // indirect
-	sigs.k8s.io/yaml v1.1.0 // indirect
-	volcano.sh/volcano v0.0.0-20190814060827-e5e4a32196c9
+	volcano.sh/volcano v0.4.0
+)
+
+replace (
+	k8s.io/api => k8s.io/api v0.16.6
```
Comment (liyinan926) [inline go.mod:39]: Which k8s version does `v0.16.6` correspond to?

File: pkg/webhook/webhook.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pkg/webhook/webhook.go:381]: yes long ago

File: go.mod (hunk 2 above)
Comment (hzxuzhonghu) [reply go.mod:39]: 1.16.6

Comment (hzxuzhonghu) [reply go.mod:39]: All the k8s related dependencies are 1.16.6 now

**Pull request merged**